cfme/cloud/instance/__init__.py

```diff
@@ -192,318 +224,594 @@ def find_quadicon(self, *args, **kwargs):
-class InstanceOwnership(CFMENavigateStep):
+class MgmtEngRelView(CloudInstanceView):
+    @View.nested
+    class form(View):  # noqa
+        server = BootstrapSelect('server_id')
+        save_button = Button('Save')
+        reset_button = Button('Reset')
+        cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        # Only the instance name is displayed, could be wrong instance on different provider
+        return False
+
+
+@navigator.register(Instance, 'EditMgmtEngineRelation')
+class EditMgmtEngineRelation(CFMENavigateStep):
+    VIEW = MgmtEngRelView
     prerequisite = NavigateToSibling('Details')
 
     # No am_i_here because the page only indicates name and not provider
+    def step(self, *args, **kwargs):
+        self.prerequisite_view.configuration.item_select('Edit Management Engine Relationship')
 
-    def step(self):
-        cfg_btn('Set Ownership')
 
+class EditTagsView(CloudInstanceView):
+    @View.nested
+    class form(View):  # noqa
+        tag_category = BootstrapSelect('tag_cat')
+        tag = BootstrapSelect('tag_add')
+        # TODO implement table element with ability to remove selected tags
+        save_button = Button('Save')
+        reset_button = Button('Reset')
+        cancel_button = Button('Cancel')
 
-@navigator.register(Instance, 'EditMgmtEngineRelation')
-class InstanceEditMgmtEngineRelation(CFMENavigateStep):
+    @property
+    def is_displayed(self):
+        # TODO match quadicon and title
+        return False
+
+
+@navigator.register(Instance, 'EditTags')
+class EditTags(CFMENavigateStep):
+    VIEW = EditTagsView
     prerequisite = NavigateToSibling('Details')
 
-    # No am_i_here because the page only indicates name and not provider
+    def step(self, *args, **kwargs):
+        self.prerequisite_view.policy.item_select('Edit Tags')
+
+
+class EvacuateView(CloudInstanceView):
+    @View.nested
+    class form(View):  # noqa
+        auto_select = CheckboxSelect('auto_select_host')
+        shared_storage = CheckboxSelect('on_shared_storage')
+        submit_button = Button('Submit')
+        cancel_button = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        # Only the instance name is displayed, cannot confirm provider
+        return False
+
+
+@navigator.register(Instance, 'Evacuate')
+class Evacuate(CFMENavigateStep):
+    VIEW = EvacuateView
+    prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        cfg_btn('Edit Management Engine Relationship')
+        try:
+            self.prerequisite_view.lifecycle.item_select('Evacuate Instance')
+        except NoSuchElementException:
+            raise DestinationNotFound('Evacuate option not available for instance')
 
 
-@navigator.register(Instance, 'AttachCloudVolume')
-class InstanceAttachVolume(CFMENavigateStep):
+class ManagePoliciesView(CloudInstanceView):
+    @View.nested
+    class form(View):  # noqa
+        policy_profiles = CheckboxSelect('protectbox')
```

Comment:
please test this because CheckboxSelect didn't work for me in provider's manage policies form.